### PR TITLE
Feature/populate staking address

### PR DESCRIPTION
**Pull request by ccali11**
@shanejearley @DemogorGod - Did this per Nate's direct request this week on one of our standup calls. 

@DemogorGod - If you want to adjust anything to fit your code style/preferences, definitely feel free, but hopefully everything I did only satisfies the feature request and/or adds clarity.

### CODE REVIEW

**Comment by ccali11** [comment]
@shanejearley - Tried to sneak in cleaning up logout in this PR, but probably would have fit better in a separate feature branch. That said, I'll keep it in here. 

@DemogorGod - Open to any and all refactors; especially where I'm initializing the two listeners in `BreakdownChart.vue`; there's probably a better design pattern to follow for that. 

I see the build failed, so I'll check that out real quick as well before I sign off today.

File: services/oracle/keyshares-1689308274.json [hunk 1]
```diff
@@ -0,0 +1 @@
...d9bdf9f3871ff5bd1c735e1df78d78d1aef669bef9e7a779e3d7fc69ed37e797dc7fdd3af5a734d1af5e6dbe36ef56f96dceb9f386dbe776dc77aebddbdedfef677b6b76baf5e774d1f6baddf75a7356fd75b73cf5d6dbe37735dbb6bae9a6bad5c7fd79c7dc71ee79d1adda73679ad797dc6f57f9d35e5fdb4e34f1d6bad9df5bf3bf746baebd7dddfad3aefcdfc75aef5e79e5bd1ef5fd1b7366fb6dcef9f3af37f34ddd71bddc7fddfdd9fdb4efce9f6dcd76e7cdb473b75ef3a69e79de34d75e77eb5d3af7b7b9dbd77575ee5df3bf3cf3beb8f5bf1f69b71b7b677477c77be9e79e6f4f387dddfb69bf387b7e1af7b77b735d1ee7bd1af766f7efdd1b75ed39f3b7f471dd9ce76edf79ed1f7dde3af74eb6db771a6f9e1a6faf776b6d346bae777bb77a7376fb75e7fbe1c6bc75cd7d6fb79f71fdfb69d75cddd75cd3ce7d77a7f9ef6e74d347f5eba6f96fa77cf5bdfaebb79c71adb971bd1df1b71c7bb738d7a77be5fe1ff38e7de1ae9eefcef8d5c7f6f3d79adde7f4f1eedae1fe3879addaefb7797b6d5a77ddb4eb5d1fe35df5ef8edce7bf5ed1e71ae74f76d38d5c79fef8d5bf7ad1f7ba7f57bad9fd5cebc","amount":"Amount of SSV tokens to be deposited to your validator's cluster balance (mandatory only for 1st validator in a cluster)","cluster":"The latest cluster snapshot data, obtained using the cluster-scanner tool. If this is the cluster's 1st validator then use - {0,0,0,0,0,false}"}},"createdAt":"2023-07-14T04:17:54.065103Z"}
```
Comment (shanejearley) [inline services/oracle/keyshares-1689308274.json:1]: @ccali11 Can you delete these files?

**Review by shanejearley — APPROVED** [review]
Looks good, fixing those new TS errors will fix build.

`npm run build --workspace @casimir/web`

**Comment by ccali11** [comment]
Ok yeah. I'll do that. Thanks.

File: apps/web/src/pages/overview/components/Staking.vue [hunk 2]
```diff
@@ -14,21 +14,28 @@ const { getEthersBalance } = useEthers()
...
 const { getCurrentPrice } = usePrice()
 
-const selectedProvider = ref<ProviderString>('')
-const selectedWallet = ref(null as null | string)
-const formattedAmountToStake = ref<string>('')
+// Staking Component Refs
 const address_balance = ref(null as null | string)
 const currentEthPrice = ref<number>(0)
 const estimatedFees = ref<number|string>('-')
+const formattedAmountToStake = ref<string>('')
```
Comment (DemogorGod) [inline apps/web/src/pages/overview/components/Staking.vue:21]: can we try to keep a consistent way of setting the type to variables (I like using the `as type` format) if you want I can try to convert to the way you have been doing it as well. Let's talk about it at some point.

File: apps/web/src/pages/overview/components/Staking.vue [hunk 3]
```diff
@@ -138,34 +139,42 @@ watch(formattedAmountToStake, async () => {
...
 })
 
-
-onUnmounted(() =>{
+onUnmounted(() => {
   window.removeEventListener('click', handleOutsideClick)
 })
 
-const stakingActionLoader = ref(false)
-const loading = ref(false)
```
Comment (DemogorGod) [inline apps/web/src/pages/overview/components/Staking.vue:159]: these variables were necessary for the front-end animation, can we change these back please, or adjust the front-end to work like it did without them?

File: apps/web/src/pages/overview/components/Staking.vue [hunk 4]
```diff
@@ -359,7 +367,7 @@ const handleDeposit = async () => {
     <button
       class="card_button  h-[37px] w-full "
```
Comment (DemogorGod) [inline apps/web/src/pages/overview/components/Staking.vue:368]: here is where the above variables are needed

**Review by DemogorGod — CHANGES_REQUESTED** [review]

File: apps/web/src/pages/overview/components/Staking.vue (hunk 2 above)
Comment (shanejearley) [reply apps/web/src/pages/overview/components/Staking.vue:21]: @DemogorGod what's been pointed out as inconsistent does not come down to matter of style. Depending on the variable we will need to use different TS features for typing. The `as` keyword is used for type assertions, which provides a specific feature to assert some new type on a variable with a prior type. For example:
    `const resp = await fetch('https://...) as CustomResponseType` (if you need to override the inferred `Response` type)

Going through this list:
    18. `const addressBalance = ref<string | null>(null)` - Provide TS `string | null` in type and initialize with null
    19. `const currentEthPrice = ref(0)` - TS infers type `number` from the initial value
    20. `const estimatedFees = ref<number | string>('-')` - Good
    21. `const formattedAmountToStake = ref('')` - TS infers type `string` from the initial value

@ccali11 how come address_balance uses snake-case?

File: apps/web/src/pages/overview/components/Staking.vue (hunk 3 above)
Comment (ccali11) [reply apps/web/src/pages/overview/components/Staking.vue:159]: I don't remember removing them. Must have been an accident (although a reference error should have thrown if I didn't have them defined but left the use of them in there)

File: apps/web/src/pages/overview/components/Staking.vue (hunk 2 above)
Comment (ccali11) [reply apps/web/src/pages/overview/components/Staking.vue:21]: That's how I found it. Made the changes. 👍

File: apps/web/src/pages/overview/components/Staking.vue (hunk 3 above)
Comment (ccali11) [reply apps/web/src/pages/overview/components/Staking.vue:159]: Oh. I didn't remove them. I just moved them up to declare all of the variables together. You can move em back I guess unless @shanejearley has a preference.

**Comment by ccali11** [comment]
Not sure how to fix the TS error in `users.ts`. Will check again tomorrow.

**Review by DemogorGod — APPROVED** [review]
All good now!

**Comment by ccali11** [comment]
Sweet. Thanks!

**Pull request merged**